packages/broker/test/unit/plugins/operator/inspectionUtils.test.ts

```diff
@@ -0,0 +1,231 @@
+            streamPart: expect.stringMatching(/^streamId#\d$/)
+        })
+    })
+
+    // TODO: few edge-cases where state changes during asynchronicity
+})
+
+describe(findNodesForTarget, () => {
+    let streamrClient: MockProxy<StreamrClient>
+    let fetchRedundancyFactorFn: jest.MockedFn<FetchRedundancyFactorFn>
+    let abortController: AbortController
+    let capturedMessageHandler: MessageListener
+    let resultPromise: Promise<NetworkPeerDescriptor[]>
+
+    beforeEach(() => {
+        streamrClient = mock<StreamrClient>()
+        streamrClient.subscribe.mockImplementation(async (_options, callback) => {
+            capturedMessageHandler = callback!
+            return mock<Subscription>()
+        })
+        fetchRedundancyFactorFn = jest.fn()
+        abortController = new AbortController()
+        resultPromise = findNodesForTarget(target, streamrClient, fetchRedundancyFactorFn, 100, abortController.signal)
+    })
+
+    afterEach(() => {
+        abortController.abort()
+    })
+
+    function comeOnline(peerDescriptor: NetworkPeerDescriptor): void {
+        capturedMessageHandler(createHeartbeatMessage(peerDescriptor), undefined as any)
+    }
+
+    it('returns empty array if no nodes found', async () => {
+        const result = await resultPromise
+        expect(result).toEqual([])
+    })
+
+    it('returns empty array if redundancy factor is undefined', async () => {
+        fetchRedundancyFactorFn.mockResolvedValueOnce(undefined)
+        comeOnline(PEER_DESCRIPTOR_ONE)
+        comeOnline(PEER_DESCRIPTOR_TWO)
+        const result = await resultPromise
+        expect(result).toEqual([])
+    })
+
+    it('returns the single node if single node found', async () => {
+        fetchRedundancyFactorFn.mockResolvedValueOnce(1)
+        comeOnline(PEER_DESCRIPTOR_ONE)
+        const result = await resultPromise
+        expect(result).toEqual([PEER_DESCRIPTOR_ONE])
+    })
+
+    it('returns one of the nodes if multiple nodes found (replicationFactor=1)', async () => {
+        fetchRedundancyFactorFn.mockResolvedValueOnce(1)
+        comeOnline(PEER_DESCRIPTOR_ONE)
+        comeOnline(PEER_DESCRIPTOR_TWO)
+        comeOnline(PEER_DESCRIPTOR_THREE)
+        const result = await resultPromise
+        expect(result.length).toEqual(1)
+        expect(result).toIncludeAnyMembers([PEER_DESCRIPTOR_ONE, PEER_DESCRIPTOR_TWO, PEER_DESCRIPTOR_THREE])
+    })
+
+    it('returns two of the nodes if multiple nodes found (replicationFactor=2)', async () => {
+        fetchRedundancyFactorFn.mockResolvedValueOnce(2)
+        comeOnline(PEER_DESCRIPTOR_ONE)
+        comeOnline(PEER_DESCRIPTOR_TWO)
+        comeOnline(PEER_DESCRIPTOR_THREE)
+        const result = await resultPromise
+        expect(result.length).toEqual(2)
+        expect(result).toIncludeAnyMembers([PEER_DESCRIPTOR_ONE, PEER_DESCRIPTOR_TWO, PEER_DESCRIPTOR_THREE])
+        expect(result[0]).not.toEqual(result[1])
```

Comment:
We always omit one of the three nodes, but maybe it it an internal implementation detail which node we omit? The `findNodesForTarget` just promises to provide some two online nodes if `replicationFactor` is 2.